File: Nancy.ViewEngines.React.UnitTests/ConfigurationTests/ScriptTest.cs
namespace Nancy.ViewEngines.React.UnitTests.ConfigurationTests
{
    using System.Linq;
    using Xunit;

    public class ScriptTest : TestBase
    {
        public ScriptTest()
            : base("Script.config")
        {
        }

        [Fact]
        public void Script_configuration_should_load_properties()
        {
            Assert.Equal("script-dir", this.Configuration.Script.Dir);
            Assert.Equal("script-name.js", this.Configuration.Script.Name);
            Assert.Equal(2, this.Configuration.Script.Extensions.Count);
            Assert.Equal("layout-name.jsx", this.Configuration.Script.Layout.Name);
        }

        [Fact]
        public void Script_configuration_should_load_extensions()
        {
            var extensions = this.Configuration.Script.Extensions
                .Cast<ReactConfiguration.ExtensionElement>()
                .Select(x => x.Name);

            Assert.Contains("es", extensions);
            Assert.Contains("jsx", extensions);
        }
    }
}

File: Nancy.ViewEngines.React.UnitTests/ConfigurationTests/ScriptTest.js
/* global describe, beforeEach, it */

const path = require('path');
const expect = require('expect.js');
const readFile = require('../../Nancy.ViewEngines.React/tools/commands/file').readFile;
const parseConfig = require('../../Nancy.ViewEngines.React/tools/commands/options').parseConfig;

describe('Script test', () => {
  beforeEach(() => {
    const file = path.resolve(__dirname, '../ConfigurationFixtures', 'Script.config');
    this.parse = readFile(file).then(parseConfig);
  });

  it('should load properties', () =>
    this.parse.then(config => {
      expect(config.script.dir).to.be('script-dir');
      expect(config.script.name).to.be('script-name.js');
      expect(config.script.extensions).to.have.length(2);
      expect(config.script.layout).to.be('layout-name.jsx');
    }));

  it('should load extensions', () =>
    this.parse.then(config => {
      expect(config.script.extensions).to.have.length(2)
        .and.to.contain('es')
        .and.to.contain('jsx');
    }));
});
